sdk/src/nonce_state.rs

```diff
@@ -141,13 +151,32 @@ impl<'a> NonceAccount for KeyedAccount<'a> {
                 if self.account.lamports < min_balance {
                     return Err(InstructionError::InsufficientFunds);
                 }
-                Meta::new()
+                Meta::new(authorized_noncer)
             }
             _ => return Err(NonceError::BadAccountState.into()),
         };
 
         self.set_state(&NonceState::Initialized(meta, recent_blockhashes[0]))
     }
+
+    fn authorize(
+        &mut self,
+        authorized_noncer: &Pubkey,
+        signers: &HashSet<Pubkey>,
+    ) -> Result<(), InstructionError> {
+        match self.state()? {
+            NonceState::Initialized(meta, nonce) => {
+                if !signers.contains(&meta.authorized_noncer) {
```

Comment:
@rob-solana should the account owner also have authority to authorize?  Would enable revocation in the non-derived address case

Reply:
create+init(Noncer)
there-after only the Noncer can/needs to sign